qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -441,58 +462,67 @@ def _run_analysis(
             )
             analysis_results.append(overview)
 
-        # Create figure and result data
         if fit_data.success:
-            # Create analysis results
-            primary_results = self._create_analysis_results(
-                fit_data=fit_data, quality=quality, **self.options.extra.copy()
+            # Add fit data to curve data table
+            fit_curves = []
+            formatted = curve_data[curve_data.format == "fit-ready"]
+            for (i, name), sub_data in list(formatted.groupby(["model_id", "model_name"])):
+                xval = sub_data.xval.to_numpy()
+                if len(xval) == 0:
+                    # If data is empty, skip drawing this model.
+                    # This is the case when fit model exist but no data to fit is provided.
+                    continue
+                # Compute X, Y values with fit parameters.
+                xval_fit = np.linspace(np.min(xval), np.max(xval), num=100)
+                yval_fit = eval_with_uncertainties(
+                    x=xval_fit,
+                    model=self._models[i],
+                    params=fit_data.ufloat_params,
+                )
+                model_fit = np.full((100, len(curve_data.columns)), None, dtype=object)
+                fit_curves.append(model_fit)
+                # xval
+                model_fit[:, 0] = xval_fit
+                # yval
+                model_fit[:, 1] = unp.nominal_values(yval_fit)
+                # yerr
+                if fit_data.covar is not None:
+                    model_fit[:, 2] = unp.std_devs(yval_fit)
+                # model_name
+                model_fit[:, 3] = name
+                # model_id
+                model_fit[:, 4] = i
+                # type
+                model_fit[:, 6] = "fit"
+            curve_data = curve_data.append_list_values(
```

Comment:
This curve_data also stores fit values. Note that we can regenerate figure object from this data, i.e. lazy figure gen (but metadata such as chisq and fit parameters must be separately provided). In follow-up maybe we can update internals of curve plotter to directly take this object.